### PR TITLE
224 new homepage

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Adding a new homepage

Todo's:
- [x] hook for stats
- [x] test mobile

About # (link your issue here)
#224 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/app/src/hooks/useTotalStats.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  const stats = useSubgraphTotalStats();
+
+  return useMemo(() => {
+    const totalDonations = stats?.collectives?.reduce((acc, collective) => acc + Number(collective.totalDonations), 0);
+    const donationsFormatted = formatGoodDollarAmount(totalDonations?.toString() ?? '0');
+
+    return {
+      totalPools: {
+        amount: stats?.activeCollectives?.length.toString() ?? '0',
+        copy: 'GoodCollective pools',
```
Comment (sirpy) [inline packages/app/src/hooks/useTotalStats.ts:26]: the text should be in the displaying component not in logic

File: packages/app/src/pages/HomePage.tsx [hunk 2]
```diff
@@ -1,110 +1,170 @@
...
+        marginBottom: 20,
+        paddingTop: 0,
+        paddingLeft: 15,
+        paddingRight: 15,
+      },
+    },
+  },
+};
+
+const HomePage = withTheme({ name: 'HomePage' })(({ containerStyles }: HomePageProps) => {
```
Comment (sirpy) [inline packages/app/src/pages/HomePage.tsx:46]: dont use withtheme or gooddesign

File: packages/app/src/theme/theme.ts [hunk 3]
```diff
@@ -2,15 +2,17 @@ import { extendTheme } from 'native-base';
 import { fontConfig, getPlatformFamilies } from '@gooddollar/good-design';
 
 import * as components from '../components/theme';
+import * as pages from '../pages/theme';
 
 export const nbTheme = extendTheme({
   fontConfig: getPlatformFamilies(fontConfig),
   colors: {
     /* g$ design system */
-    primary: '#00AFFF',
```
Comment (sirpy) [inline packages/app/src/theme/theme.ts:10]: why

File: packages/app/src/theme/theme.ts [hunk 4]
```diff
@@ -2,15 +2,17 @@ import { extendTheme } from 'native-base';
...
+    gdPrimary: '#00AFFF',
     primaryHoverDark: '#0075AC',
     white: '#FFFFFF',
     black: '#000000',
+    defaultGrey: '#F3F3F3',
```
Comment (sirpy) [inline packages/app/src/theme/theme.ts:15]: where does this come from? shouldnt if be goodgrey.50 ?

File: packages/app/src/theme/theme.ts [hunk 5]
```diff
@@ -54,7 +56,7 @@ export const nbTheme = extendTheme({
     // custom keys for breakpoints cannot be used in useBreakpoint hook so we override defaults
     base: 0,
     sm: 375,
-    md: 480,
+    md: 720,
```
Comment (sirpy) [inline packages/app/src/theme/theme.ts:59]: why?

File: packages/app/src/theme/theme.ts [hunk 6]
```diff
@@ -66,11 +68,10 @@ export const nbTheme = extendTheme({
     subheading: 'Inter',
   },
   fontSizes: {
-    '4xs': 10,
-    '2xs': 12,
-    xs: 14,
-    sm: 16,
-    md: 20,
+    '2xs': 10,
```
Comment (sirpy) [inline packages/app/src/theme/theme.ts:71]: why

File: yarn.lock [hunk 7]
```diff
@@ -12,6 +12,13 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@adraffy/ens-normalize@npm:1.10.0":
```
Comment (sirpy) [inline yarn.lock:15]: yarn diff is huge compared to small changes in package json. why?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/app/src/pages/HomePage.tsx (hunk 2 above)
Comment (L03TJ3) [reply packages/app/src/pages/HomePage.tsx:46]: so what, I copy over the hook to goodcollective?

File: yarn.lock (hunk 7 above)
Comment (L03TJ3) [reply yarn.lock:15]: updated good-design + sdk-v2

they were behind significantly

File: packages/app/src/theme/theme.ts (hunk 3 above)
Comment (L03TJ3) [reply packages/app/src/theme/theme.ts:10]: to not override the default value of primary

File: packages/app/src/theme/theme.ts (hunk 4 above)
Comment (L03TJ3) [reply packages/app/src/theme/theme.ts:15]: nope, the background grey as used on figma is not part of the goodGrey colors.
reference: https://www.figma.com/design/ihw1PxBvLxacTHnN2aj4lC/3.-Product?node-id=4761-86684&t=g5JPQQgJpTgxIYRG-4

File: packages/app/src/theme/theme.ts (hunk 5 above)
Comment (L03TJ3) [reply packages/app/src/theme/theme.ts:59]: that's the screensize from which it becomes tricky and ugly to keep the desktop layout

File: packages/app/src/theme/theme.ts (hunk 6 above)
Comment (L03TJ3) [reply packages/app/src/theme/theme.ts:71]: following design system on figma.
reference: https://www.figma.com/design/ihw1PxBvLxacTHnN2aj4lC/3.-Product?node-id=4595-86283&m=dev

File: packages/app/src/hooks/useTotalStats.ts (hunk 1 above)
Comment (L03TJ3) [reply packages/app/src/hooks/useTotalStats.ts:26]: what do we gain from separating copy from stats beyond that I have to merge the two on UI level to be displayed together?

File: packages/app/src/pages/HomePage.tsx (hunk 2 above)
Comment (sirpy) [reply packages/app/src/pages/HomePage.tsx:46]: no its not, its some hook alexey did. dont use it here. you can use native base.

File: packages/app/src/theme/theme.ts (hunk 3 above)
Comment (sirpy) [reply packages/app/src/theme/theme.ts:10]: please set the correct base theme

File: packages/app/src/theme/theme.ts (hunk 4 above)
Comment (sirpy) [reply packages/app/src/theme/theme.ts:15]: add it to the goodgrey list

File: packages/app/src/theme/theme.ts (hunk 5 above)
Comment (sirpy) [reply packages/app/src/theme/theme.ts:59]: lets make sure the breaking point is similar to what was done so far in non native base components

File: packages/app/src/hooks/useTotalStats.ts (hunk 1 above)
Comment (sirpy) [reply packages/app/src/hooks/useTotalStats.ts:26]: translation for example. and its basically just good to separate display from data and logic

Comment (L03TJ3) [reply packages/app/src/hooks/useTotalStats.ts:25]: done in this pr: https://github.com/GoodDollar/GoodCollective/pull/228/commits/0707878be975e8124aa817c05a202d40fbdc8a52

File: yarn.lock (hunk 7 above)
Comment (L03TJ3) [reply yarn.lock:15]: just to emphasise. goodcollective was using sdk's from a year ago.
resolving for now

**Pull request merged**